include/boost/beast/http/impl/fields.hpp

```diff
@@ -617,7 +617,7 @@ erase(const_iterator pos) ->
 {
     auto next = pos;
     auto& e = *next++;
-    set_.erase(e);
+    set_.erase(set_.iterator_to(e));
```

Comment:
I'm a bit disappointed that `set_.erase(e);` is allowed to compile and then malfunction

Reply:
One of the dangers of composition through inheritance.